cpg_workflows/workflow.py

```diff
@@ -499,36 +499,46 @@ def _queue_jobs_with_checks(
             and action == Action.QUEUE
             and outputs.data
         ):
-            output: str | Path
+
+            analysis_outputs: list[str | Path] = []
             if isinstance(outputs.data, dict):
-                if not self.analysis_key:
+                if not self.analysis_keys:
                     raise WorkflowError(
-                        f'Cannot create Analysis: `analysis_key` '
+                        f'Cannot create Analysis: `analysis_keys` '
                         f'must be set with the @stage decorator to select value from '
                         f'the expected_outputs dict: {outputs.data}'
                     )
-                if self.analysis_key not in outputs.data:
+                if not all(key in outputs.data for key in self.analysis_keys):
                     raise WorkflowError(
-                        f'Cannot create Analysis for stage {self.name}: `analysis_key` '
-                        f'"{self.analysis_key}" is not found in the expected_outputs '
-                        f'dict {outputs.data}'
+                        f'Cannot create Analysis for stage {self.name}: `analysis_keys` '
+                        f'"{self.analysis_keys}" is not a subset of the expected_outputs '
+                        f'keys {outputs.data.keys()}'
                     )
-                output = outputs.data[self.analysis_key]
-            else:
-                output = outputs.data
 
-            assert isinstance(output, str) or isinstance(output, Path), output
+                for analysis_key in self.analysis_keys:
+                    analysis_outputs.append(outputs.data[analysis_key])
 
-            self.status_reporter.add_updaters_jobs(
-                b=get_batch(),
-                output=output,
-                analysis_type=self.analysis_type,
-                target=target,
-                jobs=outputs.jobs,
-                prev_jobs=inputs.get_jobs(target),
-                meta=outputs.meta,
-                job_attrs=self.get_job_attrs(target),
-                update_analysis_meta=self.update_analysis_meta,
+            else:
+                analysis_outputs.append(outputs.data)
+
+            for analysis_output in analysis_outputs:
+                assert isinstance(
+                    analysis_output, (str, Path)
+                ), f'{analysis_output} should be a str or Path object'
+                self.status_reporter.add_updaters_jobs(
```

Comment:
I'm assuming that `add_updaters_job` will talk to Metamist? Instead of looping over all `analysis_outputs` here, should we allow writing multiple entries to Metamist in a batch? @illusional, any thoughts?

Reply:
@illusional, here's a link to that add_updates_job function https://github.com/populationgenomics/production-pipelines/blob/c0197bdaa0c644fa930351e525f4a84ad72326ab/cpg_workflows/status.py#L116

Reply:
Makes sense to action soon, can we raise it as an issue?